application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -2072,6 +2072,92 @@ public function invite_participants($sSessionKey, $iSurveyID )
+        if (!$this->_checkSessionKey($sSessionKey))
+            return array('status' => 'Invalid session key');
+
+        $oSurvey = Survey::model()->findByPk($iSurveyID);
+        if (!isset($oSurvey))
+            return array('status' => 'Error: Invalid survey ID');
+
+        if (Permission::model()->hasSurveyPermission($iSurveyID, 'tokens', 'update'))
+        {
+
+            if(!tableExists("{{tokens_$iSurveyID}}"))
+                return array('status' => 'Error: No token table');
+
+			$command = new CDbCriteria();
+			$command->condition = '';
+			if (count($overrideAllConditions)) {
+				foreach($overrideAllConditions as $condition)
+				{
+					$command->addCondition($condition);
+				}
+			}
+			else
+			{
+				$command->addCondition('usesleft > 0');
+				$command->addCondition("sent = 'N'");
+				$command->addCondition("remindersent = 'N'");
+				$command->addCondition("(completed ='N') or (completed='')");
+				$command->addCondition('ISNULL(validfrom) OR validfrom < NOW()');
+				$command->addCondition('ISNULL(validuntil) OR validuntil > NOW()');
+				$command->addCondition('emailstatus = "OK"');
+			}
+			$command->order = 'tid';
+
+            $aAllTokens = Token::model($iSurveyID)->findAll( $command );
+            $iAllTokensCount=count($aAllTokens);
+            unset($aAllTokens);
+
+            $iMaxEmails = (int)Yii::app()->getConfig("maxemails");
+			$command->limit = $iMaxEmails;
+			$aResultTokens = Token::model($iSurveyID)->findAll( $command );
+
+            if (empty($aResultTokens))
+                return array('status' => 'Error: No candidate tokens');
+
+            foreach($aResultTokens as $key=>$oToken)
+            {
+                //pattern taken from php_filter_validate_email PHP_5_4/ext/filter/logical_filters.c
+                $pattern = '/^(?!(?:(?:\\x22?\\x5C[\\x00-\\x7E]\\x22?)|(?:\\x22?[^\\x5C\\x22]\\x22?)){255,})(?!(?:(?:\\x22?\\x5C[\\x00-\\x7E]\\x22?)|(?:\\x22?[^\\x5C\\x22]\\x22?)){65,}@)(?:(?:[\\x21\\x23-\\x27\\x2A\\x2B\\x2D\\x2F-\\x39\\x3D\\x3F\\x5E-\\x7E]+)|(?:\\x22(?:[\\x01-\\x08\\x0B\\x0C\\x0E-\\x1F\\x21\\x23-\\x5B\\x5D-\\x7F]|(?:\\x5C[\\x00-\\x7F]))*\\x22))(?:\\.(?:(?:[\\x21\\x23-\\x27\\x2A\\x2B\\x2D\\x2F-\\x39\\x3D\\x3F\\x5E-\\x7E]+)|(?:\\x22(?:[\\x01-\\x08\\x0B\\x0C\\x0E-\\x1F\\x21\\x23-\\x5B\\x5D-\\x7F]|(?:\\x5C[\\x00-\\x7F]))*\\x22)))*@(?:(?:(?!.*[^.]{64,})(?:(?:(?:xn--)?[a-z0-9]+(?:-+[a-z0-9]+)*\\.){1,126}){1,}(?:(?:[a-z][a-z0-9]*)|(?:(?:xn--)[a-z0-9]+))(?:-+[a-z0-9]+)*)|(?:\\[(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){7})|(?:(?!(?:.*[a-f0-9][:\\]]){7,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,5})?)))|(?:(?:IPv6:(?:(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){5}:)|(?:(?!(?:.*[a-f0-9]:){5,})(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3})?::(?:[a-f0-9]{1,4}(?::[a-f0-9]{1,4}){0,3}:)?)))?(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))(?:\\.(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9]?[0-9]))){3}))\\]))$/iD';
```

Comment:
PHP's filter_var has some issues, for instance with non-ASCII in mail addresses. However, it is not likely that this is changed in the foreseeable future, because PHP7 is in feature freeze and PHP7 beta1 still uses the same regexp. For consistency within Limesurvey, it might be better to leave it as it is.
Changing it at all occurrences can be done 't.z.t'